cmd/charm/charmcmd/dockerfake_test.go

```diff
@@ -99,3 +100,133 @@ func (srv *dockerHandler) addRequest(req interface{}) {
 	defer srv.mu.Unlock()
 	srv.reqs = append(srv.reqs, req)
 }
+
+func newDockerRegistryHandler(authHandlerURL string) *dockerRegistryHandler {
+	return &dockerRegistryHandler{
+		apiVersion:     "registry/2.0",
+		authHandlerURL: authHandlerURL,
+	}
+}
+
+type registeredImage struct {
+	version1 bool
+	name     string
+	digest   string
+}
+
+type dockerRegistryHandler struct {
+	apiVersion     string
+	authHandlerURL string
+	images         []*registeredImage
+	errors         []string
+}
+
+func (srv *dockerRegistryHandler) addImage(img *registeredImage) {
+	srv.images = append(srv.images, img)
+}
+
+func (srv *dockerRegistryHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	logger.Infof("dockerRegistryHandler.ServeHTTP %v", req.URL)
```

Comment:
I'm not sure this is needed.

Reply:
not needed, but it was useful for debugging and it's only in the tests, so I think I'll leave it.